synth.py

```diff
@@ -18,25 +18,63 @@
 import synthtool.gcp as gcp
 import logging
 import subprocess
+import json
+import os
 
 logging.basicConfig(level=logging.DEBUG)
 
-gapic = gcp.GAPICGenerator()
-version = 'v1'
-library = gapic.node_library(
-        'kms', version,
-        config_path='artman_cloudkms.yaml')
-# skip index, package.json, and README.md
-s.copy(
-    library,
-    excludes=['package.json', 'README.md', 'src/index.js'],
-)
+gapic = gcp.GAPICMicrogenerator()
+versions = ['v1']
+for version in versions:
+    library = gapic.typescript_library(
+            'kms', version,
+            generator_args={
+                "grpc-service-config": f"google/cloud/kms/{version}/cloudkms_grpc_service_config.json",
+                "package-name": f"@google-cloud/kms"
+                },
+            proto_path=f'/google/cloud/kms/{version}',
+            extra_proto_files=['google/cloud/common_resources.proto', 'google/iam/v1']
+            )
+    # skip index, package.json, and README.md
+    s.copy(
+        library,
+        excludes=['package.json', 'README.md', 'src/index.ts'],
+    )
 
 # Copy common templates
 common_templates = gcp.CommonTemplates()
-templates = common_templates.node_library()
+templates = common_templates.node_library(source_location='build/src')
 s.copy(templates)
+# Remove unneeded protos in proto_list.json
+# In micro-generators, we do have default multiple common protos set as dependencies.
+# But if the proto is also specified in protoc command (used to generating client libraries), 
+# protoc command will fail becasue of the repetition and force us to remove the duplicate protos reference
+list_json='src/v1/key_management_service_proto_list.json'
+remove_proto_keywords=['/google/api', '/google/protobuf', 'google/type']
+with open(list_json, 'r') as f:
+    items=json.load(f)
+    content =[item for item in items if all([(x not in item) for x in remove_proto_keywords])]
+    new_file=json.dumps(content, indent=2) + '\n'
+with open(list_json, 'w') as f:  
+    f.write(new_file)
 
+# surgery in client.ts file
+client_file='src/v1/key_management_service_client.ts'
+s.replace(client_file, 
+'import \* as gapicConfig from \'\.\/key_management_service_client_config\.json\';',
+'import * as gapicConfig from \'./key_management_service_client_config.json\'; import {IamClient} from \'../helper\';')
+
+s.replace(client_file, 
+'private \_terminated = false;', 
+'private _terminated = false; \n   // tslint:disable-next-line no-any \n private _iamClient: any;')
```

Comment:
Don't we know the right type of IAM client? :) 

I'm sorry for fighting with `any` everywhere, but every time we disable a check we are quite possibly missing an opportunity to improve our types or even find a bug.

Reply:
totally understand it! some of them are assigned as `any` to pass the test first, but you are right, I should keep the type as I can. Thanks!

Reply:
done!